arm-templates/automate/createUiDefinition.json

```diff
@@ -127,40 +160,40 @@
               ]
             },
             "osPlatform": "Linux",
-            "count": "1"
+            "count": "1",
+            "visible": "[not(equals('Premium_LRS', steps('VirtualMachineConfig').storageAccount.type))]"
           },
           {
-            "name": "storageAccount",
-            "type": "Microsoft.Storage.StorageAccountSelector",
-            "label": "Storage account",
-            "toolTip": "Storage Account for the Virtual Machine's disks",
-            "defaultValue": {
-              "type": "Standard_LRS",
-              "name": "[concat('chefdisk', take(replace(guid(), '-', ''), 10))]"
-            },
+            "name": "vmSizePremium",
+            "type": "Microsoft.Compute.SizeSelector",
+            "label": "Virtual machine size",
+            "toolTip": "The size of virtual machine to provision.",
+            "recommendedSizes": [
+              "Standard_DS4",
+              "Standard_DS4_v2",
+              "Standard_DS5_v2"
+            ],
             "constraints": {
-              "allowedTypes": [
-                "Standard_LRS",
-                "Standard_GRS",
-                "Premium_LRS"
+              "allowedSizes": [
+                "Standard_DS4",
```

Comment:
Nope, you can use those instance types with regular storage so I think it's okay to leave them as an option.